### PR TITLE
Fix Bug 39486

### PR DESCRIPTION
**Pull request by scastria**
### Description of Change ###

Copied some row height code from ListViewRenderer into TableViewRenderer.  Verified it works with sample test project provided in bug report.

### Bugs Fixed ###

https://bugzilla.xamarin.com/show_bug.cgi?id=39486

### API Changes ###

None

### Behavioral Changes ###

None

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [ ] Rebased on top of master at time of PR
- [ ] Changes adhere to coding standard
- [ ] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Platform.iOS/Renderers/TableViewRenderer.cs [hunk 1]
```diff
@@ -103,11 +104,11 @@ protected override void OnElementPropertyChanged(object sender, PropertyChangedE
 		{
 			base.OnElementPropertyChanged(sender, e);
 
-			if (e.PropertyName == "RowHeight")
+			if (e.PropertyName == TableView.RowHeightProperty.PropertyName)
```
Comment (kingces95) [inline Xamarin.Forms.Platform.iOS/Renderers/TableViewRenderer.cs]: Nice convention!

File: Xamarin.Forms.Platform.iOS/Renderers/TableViewRenderer.cs [hunk 2]
```diff
@@ -139,12 +140,23 @@ void UpdateBackgroundView()
 
 		void UpdateRowHeight()
```
Comment (kingces95) [inline Xamarin.Forms.Platform.iOS/Renderers/TableViewRenderer.cs]: Copied from https://github.com/scastria/Xamarin.Forms/blob/ed80514659855a22b1e398af7acf66c68000c2d9/Xamarin.Forms.Platform.iOS/Renderers/ListViewRenderer.cs#L566-L566

File: Xamarin.Forms.Platform.iOS/Renderers/TableViewRenderer.cs [hunk 3]
```diff
@@ -139,12 +140,23 @@ void UpdateBackgroundView()
...
+			var rowHeight = Element.RowHeight;
+			if (Element.HasUnevenRows && rowHeight == -1 && Forms.IsiOS7OrNewer) {
+				if (Forms.IsiOS8OrNewer)
+					Control.RowHeight = UITableView.AutomaticDimension;
+			} else
+				Control.RowHeight = rowHeight <= 0 ? DefaultRowHeight : rowHeight;
+		}
 
-			Control.RowHeight = height;
+		void UpdateEstimatedRowHeight()
```
Comment (kingces95) [inline Xamarin.Forms.Platform.iOS/Renderers/TableViewRenderer.cs]: Adapted from https://github.com/scastria/Xamarin.Forms/blob/ed80514659855a22b1e398af7acf66c68000c2d9/Xamarin.Forms.Platform.iOS/Renderers/ListViewRenderer.cs#L352-L352

File: Xamarin.Forms.Platform.iOS/Renderers/TableViewRenderer.cs [hunk 4]
```diff
@@ -139,12 +140,23 @@ void UpdateBackgroundView()
 
 		void UpdateRowHeight()
 		{
-			var height = Element.RowHeight;
-
-			if (height <= 0)
-				height = 44;
+			var rowHeight = Element.RowHeight;
```
Comment (kingces95) [inline Xamarin.Forms.Platform.iOS/Renderers/TableViewRenderer.cs]: Copied from https://github.com/scastria/Xamarin.Forms/blob/ed80514659855a22b1e398af7acf66c68000c2d9/Xamarin.Forms.Platform.iOS/Renderers/ListViewRenderer.cs#L566-L566

File: Xamarin.Forms.Platform.iOS/Renderers/TableViewModelRenderer.cs [hunk 5]
```diff
@@ -148,10 +148,14 @@ public UnEvenTableViewModelRenderer(TableView model) : base(model)
 
 		public override nfloat GetHeightForRow(UITableView tableView, NSIndexPath indexPath)
 		{
-			var h = View.Model.GetCell(indexPath.Section, indexPath.Row).Height;
-			if (h == -1)
+			var cell = View.Model.GetCell(indexPath.Section, indexPath.Row);
```
Comment (kingces95) [inline Xamarin.Forms.Platform.iOS/Renderers/TableViewModelRenderer.cs]: Copied from https://github.com/scastria/Xamarin.Forms/blob/ed80514659855a22b1e398af7acf66c68000c2d9/Xamarin.Forms.Platform.iOS/Renderers/ListViewRenderer.cs#L623-L623

**Comment by kingces95** [comment]
Verified change fixes layout in (very nice) reproduction provided in bug report.

**Comment by kingces95** [comment]
:+1:

**Comment by scastria** [comment]
Thanks, I was the original submitter of the bug and of the reproduction test program.

**Comment by kingces95** [comment]
Your welcome! Thanks for your submission.

**Comment by scastria** [comment]
I also verified the following:

1. If RowHeight is set on the TableView, that RowHeight is used for all cells instead of automatic sizing
2. If HasUnevenRows is false, no automatic sizing is used
3. If different Heights are set on each Cell added to the TableView, those Heights are used instead of automatic sizing

**Comment by scastria** [comment]
How do I see which tests failed?  I click on Details and it wants a TeamCity login which I don't have.

**Comment by jassmith** [comment]
it shouldn't be running these tests... without manually running, crap...

**Comment by Vinnie1991** [comment]
I've problems with the same issue, so I would like to see a new version of xamarin.forms with this bug fix very soon!
Thanks @scastria for fixing the bug and the sample program (so I could verify that it was a problem of the framework and not by me :) )

**Comment by scastria** [comment]
what does cla-required mean? Do I need to do anything further on this to help it get accepted and merged in? I am new to pull requests.

**Pull request merged**